### ISSUE TITLE
[v4] toolbarButtons.indexOf is not a function

### ISSUE THREAD

**Issue by yellow1912** [froala/wysiwyg-editor#4262]
##### Expected behavior.
Should work like v3, or at least if there is any change then it should be reflected in the doc

##### Actual behavior.
Seem like the new code expects this to be an array (to have indexOf?) Document clearly still uses the object structure:
https://froala.com/wysiwyg-editor/docs/options/#toolbarButtons

##### Steps to reproduce the problem.
Define toolbarButtons just like the document

##### Editor version.
4.0.1

##### OS.
Any

##### Browser.
Any

`Uncaught TypeError: E.opts.toolbarButtons.indexOf is not a function`

PS: I think you guys rushed this version out of the door without enough testing, documents (on how to update) as well as demos. Even the current demo page right now has only v3 example. Perhaps a warning for people to stay back at v3 will be better?

**Comment by webaaz** [comment]
Same bug here :+1:

**Comment by frankkoenigstein** [comment]
> Same bug here 👍

👍

**Comment by EnochChoi** [comment]
> > Same bug here 👍
> 
> 👍

**Comment by webaaz** [comment]
I use https://github.com/froala/KMSFroalaEditorBundle
Other forms with React work perfectly!

@yellow1912 @frankkoenigstein @EnochChoi do you use KMSFroalaEditorBundle too?

**Comment by EnochChoi** [comment]
@webaaz I'm installing it with yarn and building it with webpack is using it.

**Comment by yellow1912** [comment]
@webaaz I use webpack (with pnpm) to build. I can share the package.json file if you need?

**Comment by gudleik** [comment]
Having same issue. Without access to source code it's even more important that the documentation is correct and updated.
Found some examples in node_modules/froala-editor/html/toolbar/ that shows how toolbarButtons should be in v4:

```js
toolbarButtons: [ ['undo', 'redo'], ['bold', 'italic', 'underline', 'strikethrough', 'subscript', 'superscript', 'outdent', 'indent', 'clearFormatting', 'insertTable', 'html'] ]
```

**Comment by zolyyy** [comment]
> Having same issue. Without access to source code it's even more important that the documentation is correct and updated.
> Found some examples in node_modules/froala-editor/html/toolbar/ that shows how toolbarButtons should be in v4:
> 
> ```js
> toolbarButtons: [ ['undo', 'redo'], ['bold', 'italic', 'underline', 'strikethrough', 'subscript', 'superscript', 'outdent', 'indent', 'clearFormatting', 'insertTable', 'html'] ]
> ```

How working in here ,,More text"?

**Comment by yellow1912** [comment]
I keep receiving emails from their marketing team talking about how great v4 is, and how it is the best of the best out there. Yet, it's been 2 weeks and we are stuck with a broken editor without enough documents. This editor has almost 700 issues, many of which have not been fixed for YEARS. Consider that it's a commercial editor that we have to pay for, I would say that it's pretty dang ****. I'm considering jumping ship.

**Comment by roggenbrot** [comment]
> I keep receiving emails from their marketing team talking about how great v4 is, and how it is the best of the best out there. Yet, it's been 2 weeks and we are stuck with a broken editor without enough documents. This editor has almost 700 issues, many of which have not been fixed for YEARS. Consider that it's a commercial editor that we have to pay for, I would say that it's pretty dang ****. 

v4 is broken like hell, new features past the customers (markdown feature is a big joke), Hidden License changes (Restribution.....) 

> I'm considering jumping ship.

+1,  although we still have a valid enterprise subscription for 1 more year.  But this release is a bad joke.

And even before 4.0 most of the updates introduced more problems for us then they solved. 

Sick of requesting the source code on every release and trying to fix broken stuff for several days or trying to do reverse engineering because of missing documentation

**Comment by JeppeKnockaert** [comment]
Not an ideal solution, but it seems to work correctly. You can still use an object in your config if you define the indexOf function on the config object like this:

```
{
  indexOf: key => Object.keys(this).indexOf(key),
  'moreText': {
    'buttons': ['bold', 'italic', 'underline', 'strikeThrough', 'subscript', 'superscript', 'fontFamily', 'fontSize']
  },
  'moreParagraph': {
    'buttons': ['alignLeft', 'alignCenter', 'formatOLSimple', 'alignRight', 'alignJustify', 'formatOL', 'formatUL']
  },
  'moreRich': {
    'buttons': ['insertLink', 'insertImage', 'insertVideo', 'insertTable', 'emoticons', 'fontAwesome']
  },
  'moreMisc': {
    'buttons': ['undo', 'redo', 'fullscreen', 'print', 'getPDF', 'spellChecker', 'selectAll', 'html', 'help'],
    'align': 'right',
    'buttonsVisible': 2
  }
}
```

**Comment by baodoan** [comment]
Same here. Too much of a rush for this version release. The quality seems low.

**Comment by mohitsharma93** [comment]
Creaer,
You change something we loss so many thing.

**Comment by detonish** [comment]
> 
> 
> @webaaz I use webpack (with pnpm) to build. I can share the package.json file if you need? Note that I'm also using vue-froala as well since my code is using vuejs 2.x

please!

**Comment by ilyaskarim** [comment]
The fix for this issue will be available in the next editor patch release 4.0.4. Thank you for your patience.

**Comment by tangix** [comment]
> I keep receiving emails from their marketing team talking about how great...

Same with Sencha ExtJS, another Idera-owned product that has been deteriorating for the last three years. 
We're still on version 7.2.0 of ExtJS and 3.2.5 with Froala. :-/

**Comment by ffjanhoeck** [comment]
@ilyaskarim When will this release happens ? We also have to stick to a schedule in our company.
Thanks is advance

**Comment by ilyaskarim** [comment]
@ffjanhoeck The release will be available by next week.

**Comment by ryanb** [comment]
Version 4.0.4 is out and fixes this issue for me. Thanks!

**Comment by johnnyoshika** [comment]
@ryanb Strange how it's not noted in the release notes. I assumed it wasn't fixed because of that: https://github.com/froala/wysiwyg-editor/releases/tag/v4.0.4

**Comment by johnnyoshika** [comment]
I can confirm that this has been fixed in v4.04

**Comment by Sergey-lang** [comment]
I also install v3 because v4 has error message "btn_more" of undefined  after resize window.

**Comment by VuyyuruRaj** [comment]
"Hi, I'm Raj the Product Marketing Manager for Froala - would love your feedback.
And BTW check out the latest V4.0.6 ( https://bit.ly/3kHelBE ) - "The most Performant Version". Thanks!

**Comment by MeghashreeAP** [comment]
Thanks for your feedback.
This issue is not reproducible or have been found to be fixed on the latest Froala release v4.0.8.
Can you please upgrade and check?

**Comment by acwr** [comment]
I can confirm this has been fixed now, thanks.

**Comment by ilyaskarim** [comment]
Fixed in V4.0.4 https://froala.com/wysiwyg-editor/changelog/#4.0.4
